cfgov/v1/atomic_elements/organisms.py

```diff
@@ -846,30 +846,112 @@ def get_context(self, value, parent_context=None):
         return context
 
 
+class VideoPlayerStructValue(blocks.StructValue):
+    @property
+    def thumbnail_url(self):
+        thumbnail_image = self.get('thumbnail_image')
+
+        if thumbnail_image:
+            return thumbnail_image.get_rendition('original').url
+
+
 class VideoPlayer(blocks.StructBlock):
-    video_url = blocks.RegexBlock(
-        label='YouTube Embed URL',
-        default='https://www.youtube.com/embed/',
-        required=True,
-        regex=r'^https:\/\/www\.youtube\.com\/embed\/.+$',
+    YOUTUBE_ID_HELP_TEXT = (
+        'Enter the YouTube video ID, which is located at the end of the video'
+        ' URL, after "v=". For example, the video ID for '
+        'https://www.youtube.com/watch?v=1V0Ax9OIc84 is 1V0Ax9OIc84.'
+    )
+
+    video_id = blocks.RegexBlock(
+        label='YouTube video ID',
+        # Set required=False to allow for non-required VideoPlayers.
+        # See https://github.com/wagtail/wagtail/issues/2665.
+        required=False,
+        # This is a reasonable but not official regex for YouTube video IDs.
+        # https://webapps.stackexchange.com/a/54448
+        regex=r'^[\w-]{11}$',
         error_messages={
-            'required': 'The YouTube URL field is required for video players.',
-            'invalid': "The YouTube URL is in the wrong format. "
-                       "You must use the embed URL "
-                       "(https://www.youtube.com/embed/video_id), "
-                       "which can be obtained by clicking Share > Embed "
-                       "on the YouTube video page.",
-        }
+            'invalid': "The YouTube video ID is in the wrong format.",
+        },
+        help_text=mark_safe(YOUTUBE_ID_HELP_TEXT)
     )
+    thumbnail_image = images_blocks.ImageChooserBlock(
+        required=False,
+        help_text=mark_safe(
+            'Optional thumbnail image to show before or after the video plays.'
```

Comment:
Should we add a second sentence here to explain what happens if you don't set this image? Something like "If the thumbnail image is not set here, the video player will show the thumbnail that was set in YouTube instead"?

Reply:
Thanks for the suggestion @niqjohnson. I've added a commit that adds this sentence. It looks like this in Wagtail:

![image](https://user-images.githubusercontent.com/654645/84780974-3f591c00-afb4-11ea-8d00-7a22d9b1850a.png)